### ISSUE TITLE
Web workers source files in web builds

### ISSUE THREAD

**Issue by leonidaspir** [playcanvas/editor#834]
We are using in our Uranus Tools for PlayCanvas a number of web workers which are instantiated using regular PlayCanvas script assets.

To keep the assets separate from any concatenation we set preload to false and that works as expected on launch window and also when exporting a build and publishing to a server of our own.

But when publishing a web build (PlayCanvas hosted build), the web worker fails to load with this error:

<img width="445" alt="image" src="https://user-images.githubusercontent.com/29543178/179707984-f0c43a29-2570-4f5e-88de-29bdbde9b8a9.png">

That happens most likely because the worker source file (script asset) is hosted on a different domain from the domain the main application is running on.

Do you think there is a way to fix this? It's pretty powerful to be able to easily spawn new web workers from regular PlayCanvas scripts:

```
const workerScript = app.assets.find('uranus-terrain.worker.js', 'script');
const worker = new Worker(workerScript.getFileUrl());
```

**Comment by yaustar** [comment]
That's interesting, looks like a cross domain issue.

Do you think you can pull together a small example project that we can test against? 

Does it work when you directly access the build from `https://s3-eu-west-1.amazonaws.com/apps.playcanvas.com/RmGD0QNB`

**Comment by leonidaspir** [comment]
Yes that url worked (build was removed so it's not accessible any more).

Actually that is a common problem with web workers and there is an easy way to work around that, by using a local blob script instead of loading the remote directly in the worker:

```
        // example code loading a cross origin worker source file
        const url = workerScript.getFileUrl();

        // --- handle launch vs build asset urls
        const workerRemoteUrl = url.indexOf('/api/assets/') === 0 ? `https://launch.playcanvas.com${url}` : url;

        // --- use a blob to load the script locally and avoid cross domain errors
        const content = `importScripts(workerRemoteUrl);`;
        const workerObjectUrl = URL.createObjectURL(new Blob([content], { type: "text/javascript" }));

        worker = new Worker(workerObjectUrl);
```
Reference: https://stackoverflow.com/questions/21913673/execute-web-worker-from-different-origin

**Comment by willeastcott** [comment]
> Feel free to close this issue since it seems it can be easily fixed from the project side.

OK, closing.
